tests/functional/adapter/dbt_clone/test_dbt_clone.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from dbt.tests.adapter.dbt_clone.test_dbt_clone import BaseClonePossible
+
+
+class TestSnowflakeClonePossible(BaseClonePossible):
```

Comment:
do we need to test `transient` and `copy_grants`?

Reply:
this admittedly is more pulled from Jeremey's draft pr where he created a snowflake implementation https://github.com/dbt-labs/dbt-core/pull/7258/files#diff-073e6ed96ac92033f0b921e061b47226b87d4f358350a1ed94fc94165f247b7aR39

while in the [docs](https://docs.snowflake.com/en/sql-reference/sql/create-clonel) doesn't specify `transient` it does mention `copy grants`

Reply:
Though it could be here has another guarantee around minimizing the history stored by the table.